File: cpp/leetcode/398.cpp
class Solution {
public:
    Solution(vector<int> nums) {
        data = nums;
    }

    int pick(int target) {
        int res =, cnts = 0;
        for (int i = 0; i < data.size(); i++) {
            if (data[i] == target) {
                cnts++;
                int j = rand() % cnts;
                if (j == 0) {
                    res = i;
                }
            }
        }
        return res;
    }

private:
    vector<int> data;
};

File: python/leetcode/398.py
class Solution(object):
    def __init__(self, nums):
        self.nums = nums

    def pick(self, target):
        from random import randint
        res, cnts = 0, 0
        for i, num in enumerate(self.nums):
            if num == target:
                cnts += 1
                idx = randint(0, i) % cnts
                if idx == 0:
                    res = i
        return res
